### ISSUE TITLE
As an user of the ownCloud iOS app, I would like to upload ALL my images to my ownCloud server

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/ios#846]
Currently the ownCloud iOS app supports different ways to upload images (and videos):

## 1. From iOS Photo app

1. Select multiple images/videos
2. Klick Sharing button
3. Select owncloud app in the iOS sharing menu

**Limitations:**

- No "Select all" button (system limitation)
- Image upload is limited to 20 items
- Video upload is limited to 3 items

**Improvements to upload all images:**

- Increase limit to 9999999… ???

## 2. From ownCloud iOS app

1. Navigate to target folder
2. Klick "•••" button on the top-right and select "Upload Photo/Video"
3. Choose Photo Album as source
4. Select many photos and videos, then click "Done" button in top-right

**Limitations:**

- No "Select all" button

**Improvements to upload all images:**

- Add "Select all" button

## 3. Using "Instant Upload" on the Settings page of the iOS app

1. Klick "Settings" in the bottom-right of the ownCloud iOS app
2. Enable "Instant Upload Photos" and "Instant Upload Videos"

**Cool:**

- Location changes can wakeup the app to constantly upload images in the background.

**Limitations:**

- Don't upload older images

**Improvements to upload all images:**

- Add config option to include older images in "Instant Upload"

@nasli @jesmrec Did I missed something? Which of the suggested improvements could be implemented? /cc @rperezb @hodyroff

**Comment by UnitedMarsupials** [comment]
> Limitations:
>  *  No "Select all" button (system limitation)

Not sure, which "system" is meant here -- iOS? ownCloud-client? ownCloud-server? If Dropbox-client can -- and does -- automatically upload all new images/videos taken since the last time it was invoked, oC should be able to do the same..

>  *  Image upload is limited to 20 items
>  *  Video upload is limited to 3 items

As I say, in #642, in my experience the limit is the _total size_ rather than the _number_ of files. For example, if your videos are small, you may be able to "share" more than 3 with ownCloud at a time.

> * Add config option to include older images in "Instant Upload"

I'd say, the "Instant Upload" options (for videos and for photos) should mean that already -- _all_ media-files of each type, regardless of age, are to be automatically synched ASAP.

Maybe, another option should limit such automatic uploads to when WiFi is available -- another thing Dropbox-client has offered for years.

**Comment by michaelstingl** [comment]
@UnitedMarsupials The paragraph "1. From iOS Photo app" is about sharing from the default Apple "Photos" app. In the "Photos" app, there is no "Select all" button. There is nothing the ownCloud app can do. It is a limitation of the iOS system "Photos" app.

The image selector in the iOS system "Photos" app has a hard limit of 20 images and 3 videos, independent from image/video size.

If you experience issues with the upload of larger files from inside the ownCloud app, you can check https://central.owncloud.org/ or [open a new issue.](https://github.com/owncloud/ios/issues/new)

There is already an open issue for "Instant Upload: "WiFi only" option": https://github.com/owncloud/ios/issues/352 . It is planned for the 3.5.5 release.

**Comment by UnitedMarsupials** [comment]
> In the "Photos" app, there is no "Select all" button. [...] There is nothing the ownCloud app can do.

ownCloud app can do, what Dropbox app does -- access the image-collection _directly_, bypassing the Photos-app. Indeed, the "Dropbox" is not even an option for image-sharing on my phone. It simply goes for the images directly upon getting started.

I urge you to give Dropbox a test-run -- the account is free up to 1Gb -- and see, what useful features can be copied by ownCloud. For my own purposes, it would've been perfect -- except it wishes to upload to Dropbox, and I want the files on my own servers.

> "WiFi only" option" [...] It is planned for the 3.5.5 release.

Awesome! Thanks.

**Comment by nasli** [comment]
**Within share option**:

> No "Select all" button (system limitation) 

>Not sure, which "system" is meant here -- iOS? ownCloud-client? ownCloud-server? If Dropbox-client can -- and does -- automatically upload all new images/videos taken since the last time it was invoked, oC should be able to do the same..

Yes as @michaelstingl said, we are outside the app, and that interface it is from Apple Photo App.
And should not be there any button select all, that is not the point to share all your camera roll through that point.

We could increase the number of files to send to OwnCloud or even unlimited, but, I would recommend this neither, there is not point to select manually hundred of photos, and later wait on that view until all have been copied to the app or uploaded. 

As you said @UnitedMarsupials , Dropbox has not an option to share files from there, you just can save the current photo, and others apps as Drive allows you to select only 10 photos and there are some reasons of limitations and performance.

For these reasons what have to be added is a new option where you can enabled and all your photos will be uploaded, and this must be done inside the OC app, in setting view could be the right place.

**Within Settings view**:
>I'd say, the "Instant Upload" options (for videos and for photos) should mean that already -- all media-files of each type, regardless of age, are to be automatically synched ASAP.

I do not agree with that, I rather another option to upload all photos. For the user case where for some time the user does not want to instant upload their photos, disabled instant upload, when the user enabled it again he expect that only will be uploaded the new ones and not the all camera photos.
We should leave instant uploads with his current behavior and add a new option to upload all the photos. Whit this we cover all use cases the user may need.

>Maybe, another option should limit such automatic uploads to when WiFi is available

It is already in our roadmap.


I will close related issues to prioritize and keep the discussion here. #642, #271

**Comment by MattiasC** [comment]
Will this functionality be replicated into the Android app as well?

**Comment by jquenee** [comment]
please... just remove this limitation. it makes no sense. even more, this limitation is not clearly shown to end user. 

Where is located into the code ? 
It should be just a matter of constant change.

**Comment by nasli** [comment]
@jquenee we have in our road map to add a new feature to upload all your photos.

In the share view there is reason for this limitation as explained here https://github.com/owncloud/ios/issues/846#issuecomment-285017529

**Comment by davivel** [comment]
Sorry, but we need to drop issues from milestone 3.6.0 to get it out next to OC 10 server.

This time we'll move topics to ```backlog``` instead of directly to next release, so that we can schedule 3.7.0 properly.

**Comment by michaelstingl** [comment]
FYI "Select all" got implemented in the successor of this ownCloud app. You can test with the TestFlight beta. We'd love to hear your feedback!

Here you can find more information:

https://owncloud.org/news/working-brand-new-ios-app-sdk/
https://central.owncloud.org/t/there-are-new-features-for-the-ios-beta-app/16992

**Comment by Majboor** [comment]
> @UnitedMarsupials The paragraph "1. From iOS Photo app" is about sharing from the default Apple "Photos" app. In the "Photos" app, there is no "Select all" button. There is nothing the ownCloud app can do. It is a limitation of the iOS system "Photos" app.
> 
> The image selector in the iOS system "Photos" app has a hard limit of 20 images and 3 videos, independent from image/video size.
> 
> If you experience issues with the upload of larger files from inside the ownCloud app, you can check https://central.owncloud.org/ or [open a new issue.](https://github.com/owncloud/ios/issues/new)
> 
> There is already an open issue for "Instant Upload: "WiFi only" option": #352 . It is planned for the 3.5.5 release.

can we have something similar to nextcloud like it automatically backup all the photos and videos

**Comment by michaelstingl** [comment]
@Majboor this is the repository for the deprecated ownCloud iOS app, which is no longer maintained. Check out the repository, for the new app:
- https://github.com/owncloud/ios-app/issues
